lib/console/gateways/gateway_identifier.ex

```diff
@@ -0,0 +1,22 @@
+defmodule Console.Gateways.GatewayIdentifier do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias Console.Gateways.Gateway
+
+  @primary_key {:id, :binary_id, autogenerate: true}
+  @foreign_key_type :binary_id
+  schema "gateway_identifiers" do
+    field :unique_identifier, :binary
+
+    has_one :gateway, Gateway, on_delete: :delete_all
+
+    timestamps()
+  end
+
+  def changeset(gateway_identifier, attrs) do
+    gateway_identifier
+    |> cast(attrs, [:unique_identifier])
+    |> validate_required([:unique_identifier])
```

Comment:
i think you can also add a unique validation here and it will return a handleable error instead of a raw database constraint violation crash